src/main/scala/sigmastate/AvlTreeData.scala

```diff
@@ -81,8 +82,7 @@ object AvlTreeData {
     AvlTreeFlags.AllOperationsAllowed,
     keyLength = 32)
 
-  def apply(avlTree: AvlTree): AvlTreeData = new AvlTreeData(ADDigest @@ avlTree.digest.toArray,
-    AvlTreeFlags.apply(avlTree.enabledOperations), avlTree.keyLength, avlTree.valueLengthOpt)
+  def apply(avlTree: AvlTree): AvlTreeData = SigmaDsl.toAvlTreeData(avlTree)
```

Comment:
Maybe remove it altogether. AvlTree is a level of API built above AvlTreeData as internal representation.
From this point of view, this method create unnatural dependency, so lower layer depends on higher one.

Reply:
Agreed. Done.